node/pkg/watchers/evm/watcher.go

```diff
@@ -151,9 +152,15 @@ func NewEthWatcher(
 func (w *Watcher) Run(ctx context.Context) error {
 	logger := supervisor.Logger(ctx)
 
+	useFinalizedBlocks := (w.chainID == vaa.ChainIDEthereum && (!w.unsafeDevMode))
 	if w.shouldCheckSafeMode {
 		if err := w.checkForSafeMode(ctx); err != nil {
-			return err
+			if !strings.Contains(err.Error(), "is not using safe mode") {
+				return err
+			}
+
+			// If the endpoint is not in safe mode, assume it is the version that supports finalized blocks.
```

Comment:
Should this be reversed?  Generally we should fall back to the safe option unless explicitly told otherwise.

Reply:
No, in this case "safe mode" refers to connecting to an additional Karura / Acala RPC layer that only publishes finalized blocks. That was a special layer they created for us. Now they want to deprecate it because their official RPC will now support finalized blocks. Once that is deployed by all the guardians, we can remove this check.

Reply:
That's kind of my point though.  If we're not using the "safe mode" endpoints then we're assuming that the endpoint supports finalized blocks.  This is a dangerous assumption if the endpoint doesn't actually support finalized blocks.

Instead we should be querying the endpoint to see if it has finalized block support and fall back to the "safe mode" endpoint if it doesn't.  Is there a version number check or something similar we can do to query the endpoint?